### PR TITLE
Graph store docs - add in a sample application and overview

### PR DESCRIPTION
**Pull request by ynx0**
This PR adds a document which provides a reference graph store application and answers common / major questions about how the application works. The reference application can be found at https://github.com/ynx0/library/.

I would encourage playing around with the app, and I'd appreciate any feedback!

**TODOs**

- [x] fix links to point to master branch once PR is merged
- [x] finalize example code, needs PR review to be completed and PR needs to be merged
- [ ] get diagram image uploaded to bucket and fix link

cc @timlucmiptev @jalehman @tacryt-socryp

### CODE REVIEW

**Comment by ynx0** [comment]
@jalehman addressed feedback, ready for another look

File: content/docs/userspace/graph-store/sample-application-overview.md [hunk 1]
```diff
@@ -0,0 +1,370 @@
++++
+title = "Sample Application Overview — Library"
```
Comment (jalehman) [inline content/docs/userspace/graph-store/sample-application-overview.md]:
```suggestion
title = "Sample Application: Library"
```

File: content/docs/userspace/graph-store/sample-application-overview.md [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+weight = 4
+sort_by = "weight"
+template = "doc.html"
+insert_anchor_links = "right"
++++
+
+
+## *Library*
```
Comment (jalehman) [inline content/docs/userspace/graph-store/sample-application-overview.md]:
```suggestion

```

**Review by jalehman — APPROVED** [review]
Sorry for the long delay on this! Got a little sidetracked. This looks complete to me—going to make a couple of small tweaks and merge it.

**Pull request merged**